src/Query/Filter/ReminderDateField.ts

```diff
@@ -0,0 +1,21 @@
+import type { Moment } from 'moment';
+import type { Task } from '../../Task';
+import { DateField } from './DateField';
+
+export class ReminderDateField extends DateField {
+    public fieldName(): string {
+        return 'reminder';
+    }
+
+    public date(task: Task): Moment | null {
+        if (task.reminders) {
+            return task.reminders.peek();
+        } else {
+            return null;
+        }
+    }
```

Comment:
### Request: Don't allow filtering by reminder date and time in the first version

- None of the filtering currently supports times, so adding 
- Even the `has due date`/`no due date` filters are unclear to me, given that this first version includes times on reminders

One way in in which this implementation of filtering is broken is that the user can only search the first reminder ...

Also, they cannot search the times on the reminder.

There will probably be others - and they all add to the complexity of what is already a really large PR.

Reply:
Agreed, The query code was a bit hectic to fully understand so could use some pointers there. I'm not really interested in supporting all the queries, mostly just a few key ones:

- Short mode (Done and working)
- has reminder date (Done and working)
- reminder after xxx (Done, only works on first reminder)

The last one I don't think is super important, most just used it `reminder after yesterday` to get a list of upcoming reminder tasks for notifications. Could probably create a separate function for this instead of using query.

Reply:
OK, for the filters that would would keep in, please do the remaining steps to complete their implementation (such as tests and documentation):

https://publish.obsidian.md/tasks-contributing/Code/How+do+I+add+a+new+task+filter

In the documentation, please be sure to record any limitations.

Reply:
https://github.com/users/Erik-Handeland/projects/2/views/1?pane=issue&itemId=26968428